### DISCUSSION TITLE
Inquiry on Robust Optimization Using YALMIP

### DISCUSSION THREAD

**Discussion by dreamersliu** [yalmip/YALMIP#1368] [Problems using YALMIP] [answered]
When I was learning robust optimization (https://yalmip.github.io/tutorial/robustoptimization/),  I encountered some Interesting problems. The code is as follows.
```
sdpvar x(3, 1) w(3,1)                                            
C = [x + w <= 10; x>= 0];                                        
W = [uncertain(w), 0 <= w, sum(w) <= 2];                  
objective = -sum(x);                                          
solver_str = 'gurobi';
ops = sdpsettings('solver',solver_str,'verbose',1,'debug',1);sol = optimize(C + W, objective, ops);                         
obj=-value(objective);                                     
x=value(x)     

```
The results are as follows. The objective is -24, and x is [8; 8; 8].

I found that the constraint sum(w)<=2 is useless, and the worst case in the results is w=[2, 2, 2]. However, the worst case should be w = [2, 0, 0], and the objective is 28 when x is [8, 10, 10]. Despite going through the documentation and various resources, I still don't understand the uncertain. 
Could you please provide some insights or point me towards resources that could assist me in better understanding this question? Any advice or recommendations would be immensely appreciated.

**Comment by johanlofberg** [comment]
If you set x(2) to 10 I simply pick w(2)=2 and thus your solution is not feasible in the worst-case.

There is no single worst-case w. The worst possible w depends on the selected solution x.

**Reply by dreamersliu** [reply]
Thanks for your answers.  If x(2) is set to 10, w(2) can't be 2. But w(1) can be set to 2, and w(3) is set to 0. In this way, sum(w) = 2. But as shown in the results obtained by the code, sum(w) is 6 which is not lower than 2, which means the constraint sum(2)<=2 is not satisfied.

**Reply by johanlofberg** [reply, marked as answer]
The other way around. w can be anything in the uncertainty set. w(2)=2 is obviously in the feasible set for W, and consequently x(2)=10 is not feasible since the constraint then is violated

You seem to fail to understand what robust optimization is. x represent the decision variables that you have to pick *first*, and then there is an opponent w which tries to make things as bad as possible to destroy for you and make constraints violated. If you pick x(2) =10, the opponent will immediately pick any w(2) > 0 causing the constraint to be violated, meaning your choice of x was not robust. Hence the only way to be robust is to have all x<=8

You want to put water in three buckets. You want to put as much total water in the buckets as possible, and they all hold 10 liters. However, there is an opponent that can bring 3 bottles of water to add to each bucket, but the total amount in the bottles combined is limited to 2 liters. The rule of the game is that if water ever touches the floor for some bucket being overfilled, you will be killed. You fill first, and then it is up to the opponent to fill up his three bottles and add to the buckets. He sees you adding 8.0001 liters to the second bucket, and thus immediately fills his second bottle with 2 liters, adds it to your second bucket, water spills over and wets the floor and you are killed. Your only solution is to never add more than 8 liters to any bucket

**Reply by dreamersliu** [reply]
Thanks for your answers which are extremely useful to me. It seems that I need to rethink what robust optimization is
